sherlock-platform/customization/src/com/google/sherlock/onboarding/EmptyProjectGenerator.java

```
@@ -0,0 +1,121 @@
+  @Override
+  public @Nullable Icon getLogo() {
+    return null;
+  }
+
+  @Override
+  public void generateProject(@NotNull Project project, @NotNull VirtualFile baseDir,
+                              @NotNull NewProjectAction.MyProjectSettings settings, @NotNull Module module) {
+    ProgressManager.getInstance().run(new Task.Modal(project, "Creating My Project", true) {
+      @Override
+      public void run(@NotNull ProgressIndicator indicator) {
+        indicator.setText("Creating project directory...");
+        createProjectDirectory(baseDir, settings);
+        indicator.setText("Adding sample files...");
+        addSampleFiles(baseDir, settings);
+        //TODO:  add project generation logic here eventually
+      }
+    });
+  }
+
+
+  @Override
+  public @NotNull ValidationResult validate(@NotNull String baseDirPath) {
+    Path baseDir = Paths.get(baseDirPath);
+    if (!Files.exists(baseDir)) {
+      return new ValidationResult("The specified directory does not exist.");
+    }
+    //Expects an empty directory
+    try (Stream<Path> files = Files.list(baseDir)) {
+      if (files.count() > 0) {
+        return new ValidationResult("The specified directory is not empty.");
+      }
+    } catch (IOException e) {
+      return new ValidationResult("Error accessing the directory: " + e.getMessage());
+    }
+    return ValidationResult.OK;
+  }
+
+  private void createProjectDirectory(@NotNull VirtualFile baseDir, @NotNull NewProjectAction.MyProjectSettings settings) {
+    try {
+      VirtualFile projectDir = baseDir.createChildDirectory(this, settings.myProjectName);
+      //TODO: create other directories and files within the project directory
+    } catch (IOException e) {
+      LOG.error("Error creating project directory", e);
+      ApplicationManager.getApplication().invokeLater(() ->
+                                                        Messages.showErrorDialog("Error creating project directory: " + e.getMessage(), "Error")
+      );
```

I think we should rethrow the exception to prevent continuing into addSampleFiles.